### PR TITLE
PP-12302 refactor DaoITestBase

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID

Refactor to use JUnit5 and RegisterExtension and not DaoITestBase.

- update all tests that used DaoITestBase to the above combo
- remove GuicedTestEnvironment and DaoITestBase as they are no longer in use

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/dao/GatewayAccountCredentialsHistoryDaoIT.java
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/dao/GatewayAccountCredentialsHistoryDaoIT.java:1]: `assertTrue` used here is a JUnit 4 assertion

File: src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java:48]: `assertEquals`, `assertNull`, and `fail` are JUnit 4 assertions

File: src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java [hunk 2]
```diff
@@ -1035,24 +999,22 @@ private TestCharge insertTestChargeWithStatus(DatabaseFixtures.TestAccount testA
...
-        DatabaseFixtures
-                .withDatabaseTestHelper(databaseTestHelper)
+        app.getDatabaseFixtures()
                 .aTestRefund()
                 .withTestCharge(defaultTestCharge)
                 .insert();
     }
 
     private void insertTestAgreement(String agreementExternalId, long gatewayAccountId) {
-        DatabaseFixtures
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java:1054]: This method doesn't appear to be used any more - is it worth just getting rid of it?

File: src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java:1]: Some more JUnit 4 assertions, `assertFalse`, `assertNotNull`, `assertNull`, and `assertTrue`

File: src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java [hunk 3]
```diff
@@ -314,40 +310,41 @@ public void shouldSaveNotificationCredentials() {
     }
 
     @Test
-    public void shouldReturnAllAccountsWhenNoSearchParamaters() {
+    void shouldReturnAllAccountsWhenNoSearchParamaters() {
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java]: Very minor but could fix this typo while we're here (`Paramaters` -> `Parameters`)

File: src/test/java/uk/gov/pay/connector/it/dao/RefundDaoJpaIT.java
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/RefundDaoJpaIT.java:1]: `assertNotNull` as above

File: src/test/java/uk/gov/pay/connector/it/dao/IdempotencyDaoIT.java [hunk 4]
```diff
@@ -17,95 +19,93 @@
...
-        Optional<IdempotencyEntity> optionalOldEntity = dao.findByGatewayAccountIdAndKey(gatewayAccount.getId(),expiringKey);
+        Optional<IdempotencyEntity> optionalOldEntity = idempotencyDao.findByGatewayAccountIdAndKey(Long.parseLong(app.getAccountId()),expiringKey);
         assertThat(optionalOldEntity.isPresent(), is(false));
 
-        Optional<IdempotencyEntity> optionalNewEntity = dao.findByGatewayAccountIdAndKey(gatewayAccount.getId(), newKey);
+        Optional<IdempotencyEntity> optionalNewEntity = idempotencyDao.findByGatewayAccountIdAndKey(Long.parseLong(app.getAccountId()), newKey);
         assertThat(optionalNewEntity.isPresent(), is(true));
     }
     
     private DatabaseFixtures.TestAccount insertTestAccount() {
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/IdempotencyDaoIT.java]: This method doesn't seem to be used anywhere -  could it be deleted?

File: src/test/java/uk/gov/pay/connector/it/dao/TokenDaoJpaIT.java [hunk 5]
```diff
@@ -20,35 +20,31 @@
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
```
Comment (DomBelcher) [inline src/test/java/uk/gov/pay/connector/it/dao/TokenDaoJpaIT.java]: As above, JUnit4

**Review by DomBelcher — COMMENTED** [review]
Just some minor changes, and a few things which could possibly be deleted

File: src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/dao/GatewayAccountCredentialsHistoryDaoIT.java
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/dao/GatewayAccountCredentialsHistoryDaoIT.java:1]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java (hunk 1 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java:48]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java (hunk 2 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/ChargeDaoIT.java:1054]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java:1]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java (hunk 3 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/GatewayAccountDaoIT.java]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/RefundDaoJpaIT.java
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/RefundDaoJpaIT.java:1]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/IdempotencyDaoIT.java (hunk 4 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/IdempotencyDaoIT.java]: 👍

File: src/test/java/uk/gov/pay/connector/it/dao/TokenDaoJpaIT.java (hunk 5 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/it/dao/TokenDaoJpaIT.java]: 👍

**Review by DomBelcher — APPROVED** [review]
Looks good 👍

**Pull request merged**